flowmanager/datasets.py

```diff
@@ -0,0 +1,90 @@
+import datetime
+import decimal
+import os
+
+import elasticsearch
+from concurrent.futures import ThreadPoolExecutor
+
+from tableschema_elasticsearch import Storage
+from datapackage_pipelines.utilities.extended_json import LazyJsonLine
+
+tpe = ThreadPoolExecutor(max_workers=1)
+
+ELASTICSEARCH_HOST = os.environ.get('EVENTS_ELASTICSEARCH_HOST', 'localhost:9200')
+DATASETS_INDEX_NAME = os.environ.get('DATASETS_INDEX_NAME', 'datahub')
+DATASETS_DOCTYPE = 'dataset'
+
+SCHEMA = {
+    'fields': [
+        {'name': 'id', 'type': 'string'},
+        {'name': 'name', 'type': 'string'},
+        {'name': 'title', 'type': 'string'},
+        {'name': 'description', 'type': 'string'},
+        {'name': 'datapackage', 'type': 'object', 'es:schema': {
+            'fields': [
+                {'name': 'readme', 'type': 'string'}
+            ]
+        }},
+        {'name': 'datahub', 'type': 'object',
+         'es:schema': {
+             'fields': [
+                 {'name': 'owner', 'type': 'string'},
+                 {'name': 'ownerid', 'type': 'string'},
+                 {'name': 'findability', 'type': 'string'},
+                 {'name': 'flowid', 'type': 'string'},
+                 {'name': 'stats', 'type': 'object', 'es:schema': {
+                    'fields': [
+                        {'name': 'rowcount', 'type': 'integer'},
+                        {'name': 'bytes', 'type': 'integer'}
+                    ]}}
+             ]}
+         },
+    ],
+    'primaryKey': ['id']
+}
+
+
+def normalize(obj):
+    if isinstance(obj, (dict, LazyJsonLine)):
```

Comment:
not needed.
which types of these are actually possible? the data written comes from a json object.